HCSP.js

```diff
@@ -0,0 +1,228 @@
+        // NOTE: "for of loop" are quite new and works only with newest browsers
+        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
+        var allItems = {};
+        for (var nextTitle of titles) {
+            allItems[nextTitle.href] = nextTitle.textContent;
+        }
+        
+        // Let the user choose the items he wants
+        var itemsSelected;
+        itemsSelected = Zotero.selectItems(allItems);
+
+        // Save these articles
+        for (var one_item in itemsSelected) {
+            articles.push(one_item)
+        }
+
+    } else {
+        // Page of the article : get the URL
+        articles = [url]
+    }
+
+    Zotero.Utilities.processDocuments(articles, scrape, function(){Zotero.done();});
+    Zotero.wait();
+}
+
+function scrape(doc, url) {
+
+    var namespace = doc.documentElement.namespaceURI;
+    // Resolver for XPaths
+
+    // init a new Zotero's item
+    var newItem = new Zotero.Item("report");
+
+    /*---------- Constants items ----------*/
+    newItem.publicationTitle = "Rapport de l'HCSP";
+    newItem.language = "fr";
+    newItem.institution = "Haut Conseil de la Santé Publique";
+    newItem.place = "Paris";
+
+    // Creator is the institution (HCSP)
+    // Names of the author are inside the PDFs (if there is one)
+    // TODO: Find a way to have a single field instead of two
+    var creator = ZU.cleanAuthor('HCSP', 'author', false);
+    newItem.creators.push(creator);
+
+    /*---------- URL ----------*/
+    newItem.url = doc.location.href;
+    
+    /*---------- Title ----------*/
+    var xpathTitle = '//li[@class="active"]';
+    var title = ZU.xpathText(doc, xpathTitle);
+    newItem.title = title;
+    Zotero.debug(title);
+ 
+    /*---------- Date ----------*/
+    // Date is not easy to find because it is not in a named bloc. It is
+    // somewhere in one of the <p> in the div 'doc-donnees'
+    // Then to find it, use a Regex on each <p> bloc
+    var xpathDate = "//div[@id='doc-donnees']/p";
+    var docDonnees = ZU.xpath(doc, xpathDate);
+
+    // test each <p> bloc (normally it's in the first one)
+    for (var nextP of docDonnees) {
+        var pContent = nextP.textContent;
+
+        if (pContent.match("Date du document")){
+            var date = pContent.match("[0-9]{2}/[0-9]{2}/[0-9]{4}");
+            break;
+        }
+    }
+    
+    newItem.date = String(date);
+
+    /*---------- Tags ----------*/
+    var xpathTags = '//div[@id="doc-donnees"]/span/a';
+    var xresultTags = ZU.xpath(doc, xpathTags);
+
+    for (var nextResult of xresultTags) {
+        var actualTag = nextResult.textContent;
+        var actualTagTrimmed = actualTag.trim();
+        newItem.tags.push(actualTagTrimmed);
+    }
+
+    /* --------- abstract -----------*/
+    var xpath_abstract = '//div[@class="avistexte"]';
+    var abstract_note = ZU.trimInternal(ZU.xpathText(doc, xpath_abstract, namespace, "\n")); 
```

Comment:
I used the argument `namespace` because I did't find how to use named argument in javascript and I wanted to access to the last argument (the separator). But there must be a more elegant solution.